core/src/main/kotlin/com/justai/jaicf/model/ScenarioModelBuilder.kt

```diff
@@ -102,4 +109,8 @@ internal class ScenarioModelBuilder {
         }
         return ScenarioModel(newStates.associateBy { it.path.toString() }, newTransitions, newHooks)
     }
-}
\ No newline at end of file
+}
+
+private fun ActivationRuleAdapter.copyOnlyIfPredicates(other: ActivationRuleAdapter) = apply {
```

Comment:
Should the mutation be that way around?
Instead of making `onlyIf` a mutable property for every `ActivationRule`, we can simply make exclusions of `any*` rules mutable and leave base class as is.
Another reason for don't making mutation on a base class level is that now we have to support a mutation of all base class properties in order to set them on `any` rules processing

Reply:
that's the right idea, thanks! i've updated this PR